runtime/executor/method_meta.cpp

```diff
@@ -233,7 +233,7 @@ bool MethodMeta::uses_backend(const char* backend_name) const {
   const auto delegates = s_plan_->delegates();
   for (size_t i = 0; i < delegates->size(); i++) {
     auto delegate = delegates->Get(i);
-    if (std::strcmp(delegate->id()->c_str(), backend_name) == 0) {
+    if (strncmp(delegate->id()->c_str(), backend_name, delegate->id()->size()) == 0) {
```

Comment:
1) nit: why drop std::?
2) this is incorrect -- it passes if the delegate ID is a prefix of the backend_name

Reply:
`std::` is an oversight. 

Prefix is a real issue, let me also check the length of the incoming in addition to strncmp.

Reply:
Thanks. Done.